ranger/core/main.py

```diff
@@ -407,8 +407,14 @@ def load_custom_commands(*paths):
         except OSError:
             LOG.debug('Unable to access plugin directory: %s', plugindir)
         else:
-            plugins = [p[:-3] for p in plugin_files
-                       if p.endswith('.py') and not p.startswith('_')]
+            plugins = []
+            for p in plugin_files:
+                if not p.startswith('_'):
+                    if p.endswith('.py'):
+                        plugins.append(p[:-3])
+                    elif os.path.isdir(os.path.join(plugindir, p)):
```

Comment:
Why do we only need the name for files but that doesn't work for directories?

Reply:
Not quite sure what you mean, but for python files, we import them using the name minus the `.py` extension, and for directories, we can just use the directory name since its treated like a module which autoloads the `__init__.py` file within it.

Reply:
Why do you need the full path for directories, while just the name suffices for files?

Reply:
Oh that's because to check if a file is a directory, we have to call `os.path.isdir` which requires a full path. While checking if a file is a python file just requires us to check if the filename is appended with `.py`.

Reply:
Ah, right, didn't consider the plugindir isn't the pwd.